src/dfcx_scrapi/core/test_cases.py

```diff
@@ -433,3 +433,153 @@ def calculate_coverage(self, coverage_type: int, agent_id: str = None):
+            credentials=self.creds, client_options=client_options
+        )
+        response =  client.calculate_coverage(request = request)
+        transitions_covered = response.transition_coverage.transitions
+
+        for transition in transitions_covered:
+            source_page = transition.source.page.display_name
+            target_page = transition.target.page.display_name
+            pages_covered.append((source_page, target_page, transition.covered))
+
+        return pages_covered
+
+    def get_covered_flows(
+        self,
+        agent_id: str = None,
+        ):
+        """
+        Get the flow wise test cases coverage information.
+
+        Args:
+            agent_id: The agent to get coverage from. Format:
+            `projects/<Project ID>/locations/<Location ID>/agents/<Agent ID>`.
+        Returns:
+            response: List of tuples having infomration of source flow,
+            target flow and coverage.
+        """
+
+        if not agent_id:
+            agent_id = self.agent_id
+
+        flows_covered = []
+
+        request = types.test_case.CalculateCoverageRequest(
+            agent=agent_id,
+            type_= "PAGE_TRANSITION",
+            )
+
+        client_options = self._set_region(agent_id)
+        client = services.test_cases.TestCasesClient(
+            credentials=self.creds, client_options=client_options
+        )
+        response =  client.calculate_coverage(request = request)
+        transitions_covered = response.transition_coverage.transitions
+
+        for transition in transitions_covered:
+            source_flow = transition.source.flow.display_name
+            target_flow = transition.target.flow.display_name
+            flows_covered.append((source_flow, target_flow, transition.covered))
+
+        return flows_covered
+
+    def get_covered_route_groups(
+        self,
+        agent_id: str = None,
+        ):
+        """
+        Get the route group wise test cases coverage information.
+
+        Args:
+            agent_id: The agent to get coverage from. Format:
+            `projects/<Project ID>/locations/<Location ID>/agents/<Agent ID>`.
+        Returns:
+            response: Dictionary with route groups as key and
+            coverage score as value.
+        """
+
+        if not agent_id:
+            agent_id = self.agent_id
+
+        route_group_covered = {}
+
+        request = types.test_case.CalculateCoverageRequest(
+            agent=agent_id,
+            type_= "TRANSITION_ROUTE_GROUP",
+            )
+        client_options = self._set_region(agent_id)
+        client = services.test_cases.TestCasesClient(
+            credentials=self.creds, client_options=client_options
+        )
+
+        response =  client.calculate_coverage(request = request)
+        routes_covered = response.route_group_coverage.coverages
+        for route in routes_covered:
+            route_group_covered[route.route_group.display_name] = route.coverage_score
```

Comment:
@hkhaitan1 consider aliasing `route.route_group.display_name` to move this under the line length limit or adding the ignore statement if you feel strongly that it should stay.